agents/smurf_file_emulator/smurf_file_emulator.py

```diff
@@ -197,41 +408,40 @@ def tune_dets(self, session, params):
                                    action_time=action_time)
 
         # Setup Notches
-        action_time = time.time()
-        files = ['channel_assignment_b0.txt', 'tune.npy']
-        for f in files:
-            self._write_smurf_file(f, 'setup_notches',
-                                   action_time=action_time)
+        sdir = self._get_action_dir('setup_notches')
+        self.tune.write_channel_assignments(basedir=sdir)
+        self.tune.write_tune(basedir=sdir)
 
         # tracking setup
         action_time = time.time()
         fname = f"{int(time.time())}.dat"
         self._write_smurf_file(fname, 'tracking_setup', prepend_ctime=False)
 
         # Short g3 stream
-        frame_gen = G3FrameGenerator(
-            self.stream_id, self.sample_rate, self.nchans
-        )
-        fname = self._get_g3_filename(frame_gen.session_id, 0, makedirs=True)
-        os.makedirs(os.path.dirname(fname), exist_ok=True)
-        session.data['noise_file'] = fname
-        writer = core.G3Writer(fname)
-        writer(frame_gen.get_obs_frame())
-        writer(frame_gen.get_status_frame())
-        start = time.time()
-        stop = start + 30
-        frame_starts = np.arange(start, stop, self.frame_len)
-        frame_stops = frame_starts + self.frame_len
-        for t0, t1 in zip(frame_starts, frame_stops):
-            if not params['test_mode']:
-                now = time.time()
-                if now < t1:
-                    time.sleep(t1 - now)
-            writer(frame_gen.get_data_frame(t0, t1))
-        writer(core.G3Frame(core.G3FrameType.EndProcessing))
+        streamer = self._new_streamer()
+        now = time.time()
+        streamer.stream_between(now, now + 30, wait=False)
+        session.data['noise_file'] = streamer.file_list[0]
 
         return True, "Wrote tune files"
 
+    def take_noise(self, session, params=None):
+        """
+        Takes a short noise timestream
+        """
```

Comment:
Agent Task docstring formatting needed. (Also on `uxm_relock` and `take_bgmap`.)